src/backend/updates.rs

```diff
@@ -64,13 +69,40 @@ fn post_upgrade() {
         || spawn(State::finalize_upgrade()),
     );
 
+    post_upgrade_fixtures();
+
     // post upgrade logic goes here
-    set_timer(Duration::from_millis(0), move || {
-        spawn(post_upgrade_fixtures());
-    });
+    // set_timer(Duration::from_millis(0), move || {
+    //     spawn(post_upgrade_fixtures());
+    // });
 }
 
-async fn post_upgrade_fixtures() {}
+fn post_upgrade_fixtures() {
+    let last_id = read(|state| state.next_post_id);
+    mutate(|state| {
+        let num_posts = Post::count(state);
+
+        for post_id in 0..last_id {
+            Post::mutate(state, &post_id, |post| {
+                post.archived = false;
+                Ok(())
+            })
+            .unwrap();
+        }
+
+        // The noop mutation above should move all posts to the heap
+        assert_eq!(state.memory.posts.len(), 0);
+        assert_eq!(state.posts.len(), num_posts);
```

Comment:
Can we also assert that allocator consists of a single segment? That should hold at this point IIUC.